automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/azure/AzureDiskSpec.scala

```diff
@@ -0,0 +1,188 @@
+          )
+          monitorGetDisk <- getDisk
+          _ = monitorGetDisk.getStatus shouldBe DiskStatus.READY
+          _ = monitorGetDisk.getName shouldBe diskName
+
+          _ <- loggerIO.info(
+            s"AzureRuntimeSpec: disk ${workspaceId}/${monitorGetDisk.getId()} in ready status detected"
+          )
+
+          _ <- loggerIO.info(s"AzureDiskSpec: About to create runtime and re-attached disk")
+
+          createReq2 = new CreateAzureRuntimeRequest()
+            .labels(labelMap)
+            .machineSize("Standard_DS1_v2")
+            .disk(
+              new AzureDiskConfig()
+                .name(diskName)
+                .size(50)
+                .labels(labelMap)
+            )
+
+          _ <- loggerIO.info(s"printing createRuntimeReq: ${createReq2.toJson}")
+
+          _ <- IO(runtimeClient.createAzureRuntime(workspaceId, runtimeName2.asString, true, createReq2))
+          _ <- loggerIO.info(s"AzureDiskSpec: Create runtime2 request submitted. Starting to poll GET")
+
+          // Verify the initial getRuntime call
+          _ <- IO.sleep(5 seconds)
+          callGetRuntime2 = IO(runtimeClient.getAzureRuntime(workspaceId, runtimeName2.asString))
+
+          intitialGetRuntimeResponse2 <- callGetRuntime2
+          _ <- loggerIO.info(s"initial get runtime response for runtime2 ${intitialGetRuntimeResponse2}")
+          _ = intitialGetRuntimeResponse2.getStatus shouldBe ClusterStatus.CREATING
+
+          _ <- loggerIO.info(
+            s"AzureDiskSpec: runtime2 ${workspaceId}/${runtimeName2.asString} in creating status detected"
+          )
+
+          _ <- loggerIO.info("AzureDiskSpec: verifying get disk2 response")
+          diskId2 = intitialGetRuntimeResponse2.getRuntimeConfig.getAzureConfig.getPersistentDiskId
+          getDisk2 = IO(diskClient.getDiskV2(diskId2.toBigInteger.intValue()))
+          diskDuringRuntimeCreate2 <- getDisk2
+          _ = diskDuringRuntimeCreate2.getStatus shouldBe DiskStatus.READY
+          _ = diskDuringRuntimeCreate2.getName shouldBe diskName
+
+          _ <- loggerIO.info(
+            s"AzureDiskSpec: disk2 ${workspaceId}/${diskDuringRuntimeCreate2.getId()} in creating status detected"
+          )
+
+          // Verify runtime 2 eventually becomes Running (in 40 minutes)
+          monitorCreateResult2 <- streamUntilDoneOrTimeout(
+            callGetRuntime2,
+            240,
+            10 seconds,
+            s"AzureDiskSpec: runtime2 ${workspaceId}/${runtimeName.asString} did not finish creating after 40 minutes"
+          )(implicitly, GeneratedLeonardoClient.runtimeInStateOrError(ClusterStatus.RUNNING))
+
+          _ <- loggerIO.info(
+            s"AzureDiskSpec: runtime2 ${workspaceId}/${runtimeName.asString} create monitor result: $monitorCreateResult2"
+          )
+          _ = monitorCreateResult2.getStatus() shouldBe ClusterStatus.RUNNING
+
+          disk2 <- getDisk2
+          _ = disk2.getStatus() shouldBe DiskStatus.READY
+
```

Comment:
Do you want to add a TODO here to test that the test file is present in the reattached PD?